File: codes/core-service/app/schemas/views/activity.py
from datetime import datetime

from pydantic import BaseModel, Field


class ActivityTimelineItemOut(BaseModel):
    id: str
    actor_user_id: str
    actor_user_id_short: str
    entity_type: str
    entity_type_label: str
    entity_id: str
    entity_id_short: str
    action: str
    metadata: dict = Field(default_factory=dict)
    created_at: datetime
    created_at_label: str


class WorkspaceActivityViewOut(BaseModel):
    workspace_id: str
    name: str
    description: str | None = None
    total_count: int
    latest_at_label: str | None = None
    items: list[ActivityTimelineItemOut]

File: codes/web/src/types/api/views/activity.ts
export type ActivityTimelineItem = {
  id: string;
  actor_user_id: string;
  actor_user_id_short: string;
  entity_type: string;
  entity_type_label: string;
  entity_id: string;
  entity_id_short: string;
  action: string;
  metadata: Record<string, unknown>;
  created_at: string;
  created_at_label: string;
};

export type WorkspaceActivityView = {
  workspace_id: string;
  name: string;
  description?: string | null;
  total_count: number;
  latest_at_label?: string | null;
  items: ActivityTimelineItem[];
};
